pkg/kubectl/cmd/cmd.go

```diff
@@ -331,7 +332,16 @@ func DefaultClientConfig(flags *pflag.FlagSet) clientcmd.ClientConfig {
 
 func checkErr(err error) {
```

Comment:
Go has a pattern for handling errors.  We should be using it...  The command execution functions not returning an error is a quirk of commander, not a behavior we should be mimicking.

----- Original Message -----
> > @@ -331,7 +332,16 @@ func DefaultClientConfig(flags *pflag.FlagSet)
> > clientcmd.ClientConfig {
> >  
> >  func checkErr(err error) {
> 
> The abstraction of functionality to handle an error is not a problem - the
> problem is calling this functionality from all the various functions from
> which we call it. I added some comments on #2146 (the issue that I think
> you're referring to) to clarify. In this case we're abstracting out the
> program printing a specific kind of string and then fataling on error. As
> long as that's the same functionality that the util function provides, then
> it's okay to merge them, but I don't think having two of these functions is
> intrinsically an issue. Probably the naming could be improved...
> `fatalOnError` would be better.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/5016/files#r25804575